hedera-node/src/main/java/com/hedera/services/fees/calculation/UsageBasedFeeCalculator.java

```diff
@@ -128,6 +133,54 @@ public FeeObject estimateFee(SignedTxnAccessor accessor, JKey payerKey, StateVie
 		return feeGiven(accessor, payerKey, view, prices, exchange.rate(at));
 	}
 
+	@Override
+	public long activeGasPriceInTinybars() {
+		return gasPriceInTinybars(usagePrices.activePrices(), exchange.activeRate());
+	}
+
+	@Override
+	public long estimatedGasPriceInTinybars(HederaFunctionality function, Timestamp at) {
+		var rates = exchange.rate(at);
+		var prices = usagePrices.pricesGiven(function, at);
+		return gasPriceInTinybars(prices, rates);
+	}
+
+	@Override
+	public long estimatedNonFeePayerAdjustments(SignedTxnAccessor accessor, Timestamp at) {
+		switch (accessor.getFunction()) {
+			case CryptoCreate:
+				var cryptoCreateOp = accessor.getTxn().getCryptoCreateAccount();
+				return -cryptoCreateOp.getInitialBalance();
+			case CryptoTransfer:
+				var payer = accessor.getPayer();
+				var cryptoTransferOp = accessor.getTxn().getCryptoTransfer();
+				var adjustments = cryptoTransferOp.getTransfers().getAccountAmountsList();
+				long cryptoTransferNet = 0L;
+				for (AccountAmount adjustment : adjustments) {
+					if (payer.equals(adjustment.getAccountID())) {
+						cryptoTransferNet += adjustment.getAmount();
+					}
+				}
+				return cryptoTransferNet;
```

Comment:
Should this be ```return -cryptoTransferNet;```?

Reply:
For a `CryptoTransfer`, we can directly compute the affect on the payer (for example, if it sends `1_000 tinybar`, then `cryptoTransferNet=-1000`, or if it receives `500 tinybar`, then `cryptoTransferNet=+500`).

For the other operation types, we need to multiply a quantity by -1 to get the `cryptoTransferNet`, since in these cases there is some amount (`gas`, an `initialBalance`, or a `sent` amount) that will be **deducted from the payer**.

Reply:
Yeah. I realized that all amount in transfer list from this payer is negative. Thanks for the clarification.